test/blocks/modals/modals.test.js

```diff
@@ -170,4 +170,10 @@ describe('Modals', () => {
     dialogmodal.classList.remove('commerce-frame');
     expect(window.innerWidth).not.equal(dialogmodal.offsetWidth);
   });
+
+  it('does not error for a modal with a non-querySelector compliant hash', () => {
+    // Test passing, means there was no error thrown
+    window.location.hash = '#milo=&';
```

Comment:
Should we not `await` for any modal logic to execute after calling this?

Reply:
Hmm, I tested this, and it should work but given event listeners are async let's make the test more resilient and wait for the hashchange event listener to be triggered.

We can't really expect an event listener to not throw, but if it does throw - the test will fail.